datasets/ONS-International-exports-of-services-from-subnational-areas-of-the-UK/main.py

```diff
@@ -17,106 +17,279 @@
+trace.Export_Services("Defined from cell ref A5 down")
+
+origin = cell.shift(0,-1).fill(RIGHT).is_not_blank().is_not_whitespace()  
+trace.Service_Origin_Geography("Defined from cell ref A6 across")
+
+destination = cell.fill(RIGHT).is_not_blank().is_not_whitespace() 
+trace.Service_Destination("Defined from cell ref A5 across")
+
+observations = destination.fill(DOWN).is_not_blank().is_not_whitespace() 
+dimensions = [
+            HDim(industry,'Export Services',DIRECTLY,LEFT),
+            HDim(destination, 'Service Destination',DIRECTLY,ABOVE),
+            HDim(origin, 'Service Origin Geography',CLOSEST,LEFT),    
+]  
+tidy_sheet = ConversionSegment(tab, dimensions, observations)   
+savepreviewhtml(tidy_sheet, fname=tab.name + "Preview.html")
+trace.with_preview(tidy_sheet)
+#store 4b and appending to combined_dataframe
+trace.store("combined_dataframe", tidy_sheet.topandas())
 # -
 
-next_table['Export Services'].unique()
+#post processing
+df = trace.combine_and_trace(datasetTitle, "combined_dataframe")
+df
+
+df.rename(columns={'OBS' : 'Value','DATAMARKER' : 'Marker'}, inplace=True)
+
+#mapping Service Origin Geography outputs to nuts1 codes 
+trace.Service_Origin_Geography("mapping Service Origin Geography outputs to nuts1 nuts2 and nuts3 codes ")
+df['Service Origin Geography'] = df['Service Origin Geography'].map(
+    lambda x: {  
+'United Kingdom':'nuts1/all',
+'North East ':'nuts1/UKC',
+'North West':'nuts1/UKD',
+'Yorkshire and The Humber':'nuts1/UKE',
+'East Midlands':'nuts1/UKF',
+'West Midlands':'nuts1/UKG',
+'East of England':'nuts1/UKH',
+'London':'nuts1/UKI',
+'South East':'nuts1/UKJ',
+'South West':'nuts1/UKK',
+'Wales':'nuts1/UKL',
+'Scotland':'nuts1/UKM',
+'Northern Ireland':'nuts1/UKN'      
+        }.get(x, x))
+
+f1=(df['NUTS'] =='nuts2/')
+df.loc[f1,'Service Origin Geography'] = 'nuts2/' + df.loc[f1,'Service Origin Geography']
 
+f1=(df['NUTS'] =='nuts3/')
+df.loc[f1,'Service Origin Geography'] = 'nuts3/' + df.loc[f1,'Service Origin Geography']
 
+df['Period'] = "year/2017"
+trace.Period("Hard coded as year/2017")
+
+df['Export Services'] = df['Export Services'].apply(pathify)
+df
+
+df["Service Origin Geography"] = df["Service Origin Geography"].apply(pathify)
+
+df["Service Destination"] = df["Service Destination"].apply(pathify)
+
+df['Service Destination'] = df['Service Destination'].map(
+    lambda x: { 'total' : 'all', 'row' :'rest-of-world'}.get(x, x))
+
+df['Marker'] = df['Marker'].map(lambda x: { '..' : 'suppressed' }.get(x, x))
+
+df['Flow Directions'] = 'exports'
+trace.Flow_Directions("Adding Flow directions column to dataframe with value exports")
+
+#define the column order output into dataframe called tidy
+tidy = df[['Period','Export Services','Service Origin Geography','Service Destination', 'Flow Directions', 'Value','Marker']]
+tidy
+
+#output cube and spec
+cubes.add_cube(scraper, tidy.drop_duplicates(), "ONS-International-exports-of-services-from-subnational-areas-of-the-UK" )
```

Comment:
change to pass the distribution title also e.g : 
`cubes.add_cube(scraper, tidy.drop_duplicates(), distribution.title )`

Reply:
Used distribution.title() instead of hardcoding the title